share/functions/__fish_describe_command.fish

```diff
@@ -6,15 +6,34 @@
 # The whatis database is non-existent, so apropos tries (and fails) to create it every time,
 # which takes about half a second.
 #
-# So we disable this entirely in that case.
-if test (uname) = Darwin
-    set -l darwin_version (uname -r | string split .)
-    # macOS 15 is Darwin 19, this is an issue up to and including 10.15.3.
-    if test "$darwin_version[1]" = 19 -a "$darwin_version[2]" -le 3
-        function __fish_describe_command
+# Instead, we build a whatis database in the user cache directory
+# and override the MANPATH using that directory before we run `apropos`
+#
+# the cache is rebuilt one a day
+#
+function __fish_apropos
+    if test (uname) = Darwin
+        set -l cache $HOME/.cache/fish/
+        if test -n "$XDG_CACHE_HOME"
+            set cache $XDG_CACHE_HOME/fish
+        end
+
+        set -l db $cache/whatis
+        set -l max_age 86400 # one day
+        set -l age $max_age
+
+        if test -f $db
+            set age (math (date +%s) - (stat -f %m $db))
+        end
+
+        if test $age -ge $max_age
+            echo "making cache $age $max_age"
+            mkdir -m 700 -p $cache
+            man --path | string split : | xargs /usr/libexec/makewhatis -o $db >/dev/null 2>&1
```

Comment:
Nooooooooooooooooo!

Remove the `xargs`, `xargs` splits on whitespace, so this will break if you have a directory with spaces!

Just pass it via a command substitution:

```fish
/usr/libexec/makewhatis -o $db (man --path | string split :) >/dev/null 2>&1
```

Reply:
Fixed it. Sorry about that